src/mainloop.rkt

```diff
@@ -139,32 +139,36 @@
   (timeline-event! 'localize)
 
   (define loc-errss
-    (batch-localize-error (map alt-expr (^next-alts^)) (*context*)))
-  (define repr (context-repr (*context*)))
+     (batch-localize-error (map alt-expr (^next-alts^)) (*context*)))
+  (define loc-costss
+     (batch-localize-cost (map alt-expr (^next-alts^)) (*context*)))
+    (define repr (context-repr (*context*)))
 
   ; high-error locations
   (^locs^
-    (for/list ([loc-errs (in-list loc-errss)]
-               #:when true
-               [(err expr) (in-dict loc-errs)]
-               [i (in-range (*localize-expressions-limit*))])
-      (timeline-push! 'locations (~a expr) (errors-score err)
-                      (not (patch-table-has-expr? expr)) (~a (representation-name repr)))
-      expr))
-
-  ; low-error locations (Pherbie-only with multi-precision)
-  (^lowlocs^
-    (if (and (*pareto-mode*) (not (null? (platform-conversions (*active-platform*)))))
+    (remove-duplicates 
+      (append 
         (for/list ([loc-errs (in-list loc-errss)]
-                   #:when true
-                   [(err expr) (in-dict (reverse loc-errs))]
-                   [i (in-range (*localize-expressions-limit*))])
-          (timeline-push! 'locations (~a expr) (errors-score err) #f (~a (representation-name repr)))
-          expr) 
-        '()))
+                  #:when true
+                  [(err expr) (in-dict loc-errs)]
+                  [i (in-range (*localize-expressions-limit*))])
+                  (timeline-push! 'locations (~a expr) (errors-score err)
+                          (not (patch-table-has-expr? expr)) (~a (representation-name repr)))
+          expr)
+          
+        (for/list ([locs-costs (in-list loc-costss)]
+                  #:when true
+                  [pair (in-list locs-costs)]
+                  [i (in-range (*localize-expressions-limit*))])
+                  (car pair)))))
```

Comment:
Also add timeline data for `loc-costs`. Like I said you can pretty much copy and paste the first block for the second block